docs/lib/restapi/fragments/native_common/authz/common.md

```diff
@@ -0,0 +1,124 @@
+
+Your requests will be automatically have the IAM credentials when your API is set up to use IAM as the authorization mode. For example, you can check this by looking at what authorization type is configured for the API
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "IAM"
+        }
+    }
+}
+```
+
+## API Key
+
+If you want to configure the REST API to be a public API, you can set an API key on the API Gateway and update your configuration to use the API key. You should have the following configuration in your `amplifyconfiguration.json` file containing the `API_KEY` authorization mode and the value of the API key to be used when creating requests.
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "API_KEY",
+            "API_KEY": "[API_KEY]"
+        }
+    }
+}
+```
+
+
+## Cognito User pool authorization
+
+If you set up the API Gateway with a custom authorization with a Cognito User pool, then you can set the `amplifyconfiguration.json` to use `COGNITO_USER_POOL`. 
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "COGNITO_USER_POOL"
+        }
+    }
+}
+```
+
+Also in `amplifyconfiguration.json` file, under `awsCognitoAuthPlugin` will contain the corresponding Cognito configuration values that refer to the Cognito User Pool you want to use.
+```json
+{
+    "CognitoUserPool": {
+        "Default": {
+            "PoolId": "POOL-ID",
+            "AppClientId": "APP-CLIENT-ID",
+            "AppClientSecret": "APP-CLIENT-SECRET",
+            "Region": "us-east-1"
+        }
+    },
+    "CredentialsProvider": {
+        "CognitoIdentity": {
+            "Default": {
+                "PoolId": "YOUR-COGNITO-IDENTITY-POOLID",
+                "Region": "us-east-1"
+            }
+        }
+    }
+}
+```
+
+This will automatically add the Authorization header with access token to your requests to this API. To see more details on how to configure the API Gateway with the custom authorization, see [this](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-integrate-with-cognito.html)
+
+
+## Custom Interceptor
+
+If you have a custom token from another provider, you can set it `custom` and create your own interceptor
+
+```
+CUSTOM INTERCEPTOR CODE
+```
```

Comment:
Custom interceptor is not available in Android right now. This section should perhaps be moved to iOS-only doc?

Reply:
Hi @raphkim, custom interceptors should fall into the use cases of
1. Adding metadata to requests, including having customizations on top of Cognito User Pool as a provider
2. Retrieving a token from a provider other than cognito user pool.  

I wonder what is currently supported in Android, 
how do I update request header values? Do i simply add to the APIRequest object?
how do I add a custom OIDC auth provider? (I see a case for OIDC [here](https://github.com/aws-amplify/amplify-android/blob/main/aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncSigV4SignerInterceptorFactory.java#L105-L117))

Reply:
on iOS, it currently looks like this https://github.com/aws-amplify/amplify-ios/issues/780#issuecomment-702926246

Reply:
* You use `RestOptions` to add headers to REST API requests
```
RestOptions.builder()
    .addHeader("key", "value")
    .build();
```
* When constructing `AWSApiPlugin`, you can pass optional `ApiAuthProviders` instance alongside it. So the code will look like this:
```
new ApiAuthPlugin(ApiAuthProviders.builder()
    .oidcAuthProvider(() -> "<token here>")
    .build());
```
The lambda can be replaced with an actual implementation of `OidcAuthProvider` as well.

Reply:
- Updated iOS code snippet to use OIDC provider.
- added section for using Cognito user pool as _the_ OIDC provider, and retrieving the ID Token. (since we default to use accessToken)
- added placeholder for Android's example
```java
ApiAuthProviders authProviders = ApiAuthProviders.builder()
    .oidcAuthProvider(() -> {
        // TODO: Amplify.Auth.fetchAuthSession
    })
    .build();
```